### PR TITLE
[Enhancement] Optimize parquet scanner by arrow io coalesce

### PR DESCRIPTION
**Pull request by wyb**
## What I'm doing:
```
performance test 0: tpcds store_sales, 23 columns, one file 649M, 7218819 lines
                 | file read time | file read count | memory
io coalesce 8M   | 13s            |   80            | 587M
buffer stream 8M | 15s            |   176           | 1313M
buffer stream 1M | 29s            |   1145          | 1157M

performance test 1: 1001 columns table, one file 147M, 50000 lines
                 | file read time | file read count | memory
io coalesce 8M   | 3s             |   20            | 1G
buffer stream 8M | 15s            |   1003          | 10.1G
buffer stream 1M | 15s            |   1003          | 3.3G
```

Compared with buffer stream, io coalesce achieves higher performance, uses less memory, especially when there are many columns.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

File: be/src/exec/parquet_reader.cpp [hunk 1]
```diff
@@ -85,6 +83,28 @@ Status ParquetReaderWrap::_init_parquet_reader() {
         */
         arrow_reader_properties.set_coerce_int96_timestamp_unit(arrow::TimeUnit::MICRO);
 
+        // default batch size is 64K
+        arrow_reader_properties.set_batch_size(config::arrow_read_batch_size);
```
Comment (mofeiatwork) [inline be/src/exec/parquet_reader.cpp:87]: why not change the default batch size ?

Comment (wyb) [reply be/src/exec/parquet_reader.cpp:88]: this is arrow default batch size,
the bigger batch size, the more memory it uses.

**Review by murphyatwork — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Comment by wyb** [comment]
https://github.com/Mergifyio backport branch-3.0

**Comment by wyb** [comment]
https://github.com/Mergifyio backport branch-2.5

**Pull request merged**